src/lib/Lib/AbsoluteCompilationDatabase.cpp

```diff
@@ -51,14 +51,22 @@ static
 std::vector<std::string>
 adjustCommandLine(
     const std::vector<std::string>& cmdline,
-    const std::vector<std::string>& additional_defines)
+    const std::vector<std::string>& additional_defines,
+    std::unordered_map<std::string, std::vector<std::string>> const& includePathsByCompiler)
 {
     std::vector<std::string> new_cmdline;
     std::vector<std::string> discarded_cmdline;
```

Comment:
Does `discarded_cmdline` affect the output at all? Looks like an unused variable. Unless I didn't understand it, references from cmdline are inserted there and there it goes out of scope unused.

(I know that's not you're code but it's worth commenting on this kind of thing in passing to improve things)